Realm/RLMArray.mm

```diff
@@ -437,3 +438,23 @@ - (instancetype)reversedSortDescriptor {
 }
 
 @end
+
+@interface RLMArray (Handover) <RLMThreadConfined_Private>
+@end
+
+@implementation RLMArray (Handover)
+
+- (realm::AnyThreadConfined)rlm_handoverData {
+    @throw RLMException(@"Unexpected handover of unmanaged `RLMArray`");
```

Comment:
I believe that an `RLMArray` property of a currently unmanned object is of this type rather than the `RLMArrayLinkView` subclass. I guess the handover code should first notice that `realm` is `nil`, and thus not call this code, so it would be a bug in our code. I'll replace with assertions.